src/app/pages/camera.cpp

```diff
@@ -27,25 +27,91 @@ CameraPage::CameraPage(QWidget *parent) : QWidget(parent)
+    connect(this, &CameraPage::connected_network, [layout, this]() { layout->setCurrentIndex(2); this->connected=true; });
+
+
+    videoContainer_ = nullptr;
+    videoWidget_ = nullptr;
+
+    // Write camera_overlay to /tmp so that we can later point gstreamer to it
+    if (QFile::exists("/tmp/dash_camera_overlay.svg"))
+    {
+        QFile::remove("/tmp/dash_camera_overlay.svg");
+    }
+    QFile::copy(":/camera_overlay.svg", "/tmp/dash_camera_overlay.svg");
+}
 
-    if (this->config->get_cam_autoconnect())
+void CameraPage::showEvent(QShowEvent *event)
+{
+    QWidget::showEvent(event);
+    DASH_LOG(info) << "[CameraPage] Show event.";
+    if (this->config->get_cam_autoconnect() && this->connected==false)
+    {
         this->connect_cam();
+    }
+}
+
+void CameraPage::init_gstreamer_pipeline(std::string vidLaunchStr_, bool sync)
+{
+    videoWidget_ = new QQuickWidget(videoContainer_);
+
+    surface_ = new QGst::Quick::VideoSurface;
+    videoWidget_->rootContext()->setContextProperty(QLatin1String("videoSurface"), surface_);
+    videoWidget_->setSource(QUrl("qrc:/camera_video.qml"));
+    videoWidget_->setResizeMode(QQuickWidget::SizeRootObjectToView); 
+
+    videoSink_ = surface_->videoSink();
+
+    GError* error = nullptr;
+    std::string vidLaunchStr = vidLaunchStr_;
+    if(this->config->get_cam_overlay()){
+        double width = this->config->get_cam_overlay_width()/100.0;
+        double height = this->config->get_cam_overlay_height()/100.0;
+        double x = (1-width)/2.0;
+        double y = (1-height);
+        vidLaunchStr = vidLaunchStr + 
+                    " ! videoconvert ! rsvgoverlay location=/tmp/dash_camera_overlay.svg width-relative="+std::to_string(width)+" height-relative="+std::to_string(height)+" x-relative="+std::to_string(x)+" y-relative="+std::to_string(y);
+    }
+    vidLaunchStr = vidLaunchStr +
+                    " ! videoconvert " +
+                    " ! capsfilter caps=video/x-raw name=mycapsfilter";
+    DASH_LOG(info) << "[CameraPage] Created GStreamer Pipeline of `"<<vidLaunchStr<<"`";
+    vidPipeline_ = gst_parse_launch(vidLaunchStr.c_str(), &error);
+    GstBus* bus = gst_pipeline_get_bus(GST_PIPELINE(vidPipeline_));
+    gst_bus_add_watch(bus, (GstBusFunc)&CameraPage::busCallback, this);
+    gst_object_unref(bus);
+
+    GstElement* sink = QGlib::RefPointer<QGst::Element>(videoSink_);
+    g_object_set(sink, "force-aspect-ratio", false, nullptr);
+    g_object_set(sink, "sync", sync, nullptr);
+
+    g_object_set(sink, "async", false, nullptr);
+
+    GstElement* capsFilter = gst_bin_get_by_name(GST_BIN(vidPipeline_), "mycapsfilter");
+    gst_bin_add(GST_BIN(vidPipeline_), GST_ELEMENT(sink));
+    gst_element_link(capsFilter, GST_ELEMENT(sink));
 }
 
+
 QWidget *CameraPage::connect_widget()
 {
     QWidget *widget = new QWidget(this);
     QVBoxLayout *layout = new QVBoxLayout(widget);
 
+    QPushButton *settings_button = new QPushButton(this);
```

Comment:
assuming this is the settings stuff you had problems with? Go ahead and remove everything related to that, I should be coming out with a unified way of having per-page settings soon enough